### PR TITLE
Bump Fleece and add test for lack of precision in some math results

### PR DESCRIPTION
**Pull request by borrrden**
See CBL 978

### CODE REVIEW

**Comment by borrrden** [comment]
Most of the work here is downstream in Fleece

File: LiteCore/Query/SQLitePredictionFunction.cc [hunk 1]
```diff
@@ -108,6 +108,14 @@ namespace litecore {
         return (i1.count() == i2.count());
     }
 
+    static void writeResultNumeric(sqlite3_context* ctx, double num){
```
Comment (snej) [inline LiteCore/Query/SQLitePredictionFunction.cc:111]: I don't think this is necessary, since these results are unlikely to be integral.

Comment (borrrden) [reply LiteCore/Query/SQLitePredictionFunction.cc:111]: More than half of our assertions on this method use integral results.  Is it ok to change all of them to float instead (query distance functions)

Comment (snej) [reply LiteCore/Query/SQLitePredictionFunction.cc:111]: Sure.

**Comment by borrrden** [comment]
Latest failure is due to another issue being investigated.  However I noticed there are N1QL parser tests that fail that don't seem to be included in the Cmake project.  Need clarification on if they are needed.

**Comment by snej** [comment]
N1QLParserTest.cc is in the CppTests target in the Xcode build, so yeah, it should be in CMake too.

**Comment by borrrden** [comment]
@snej Take a look at the latest commit for the N1QL fixes

**Pull request merged**